var/exp_repo/experiments/amg2023/experiment.py

```diff
@@ -0,0 +1,175 @@
+                                "exclude": excludes,
+                                m_tag: matrices,
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    def compute_applications_section(self):
+        if self.spec.satisfies("workload=problem1"):
+            self.workload = "problem1"
+        else:
+            self.workload = "problem2"
+
+        if self.spec.satisfies("experiment=example"):
+            return self.make_experiment_example();
+        elif self.spec.satisfies("experiment=strong"):
+            return self.make_experiment_strong();
+        elif self.spec.satisfies("experiment=weak"):
+            return self.make_experiment_weak();
+        else:
+            raise NotImplementedError(
+                "Unsupported experiment. Only strong, weak and example experiments are supported"
+            )
+
+    def compute_spack_section(self):
+        app_name = self.spec.name
+
+        app_spack_spec = "amg2023@develop +mpi{modifier_spack_variant}"
+        hypre_spack_spec = "hypre@2.31.0 +mpi+mixedint~fortran{modifier_spack_variant}"
+
+        #TODO: Handle compiler handles through system.py
+        compiler = "default-compiler"
+
+        package_specs = {}
+        #TODO: Handle package handles through system.py
+        packages = ["default-mpi", 'lapack', "hypre", app_name, "{modifier_package_name}"]
+
+        if self.spec.satisfies("programming_model=openmp"):
+            app_spack_spec += "+openmp"
+            hypre_spack_spec += "+openmp"
+        elif self.spec.satisfies("programming_model=cuda"):
+            app_spack_spec += "+cuda cuda_arch={cuda_arch}"
+            hypre_spack_spec += "+cuda cuda_arch={cuda_arch}"
+            #TODO: Handle package handles through system.py
+            packages = ["cuda", "cublas-cuda"] + packages
+            cuda_spack_spec = "cuda@{default_cuda_version}+allow-unsupported-compilers"
+            package_specs["cuda"] = {"pkg_spec": cuda_spack_spec,
+                                      "compiler": compiler,}
+
+        elif self.spec.satisfies("programming_model=rocm"):
+            app_spack_spec += "+rocm amdgpu_target={rocm_arch}"
+            hypre_spack_spec += "+rocm amdgpu_target={rocm_arch}"
+            #TODO: Handle package handles through system.py
+            packages = ["blas-rocm"] + packages
+
+        package_specs["hypre"] = {"pkg_spec": hypre_spack_spec,
+                                  "compiler": compiler,}
+
+        package_specs[app_name] = {"pkg_spec": app_spack_spec,
+                                   "compiler": compiler,}
+
+        return {
+            "packages": package_specs,
+            "environments": {app_name: {"packages": packages}},
+        }
+
+    def compute_ramble_dict(self):
+        ramble_dict = super(Amg2023, self).compute_ramble_dict()
+        ramble_dict["ramble"]["software"] = ramble_dict["ramble"].pop("spack")
```

Comment:
@pearce8 @scheibelp @becker33 This will be removed once the base experiment.py is fixed